gramps/gen/filters/_genericfilter.py

```diff
@@ -144,51 +147,29 @@ def apply_logical_op_to_all(
     ):
         final_list = []
 
-        optimizer = Optimizer(self)
-        handles_in, handles_out = optimizer.get_handles()
+        optimizer = Optimizer(set(self.get_all_handles(db)), self)
+        handles_in = optimizer.get_handles()
 
         LOG.debug(
             "Optimizer handles_in: %s",
-            len(handles_in) if handles_in is not None else None,
+            len(handles_in),
         )
-        LOG.debug("Optimizer handles_out: %s", len(handles_out))
         if id_list is None:
```

Comment:
I think there is a further optimization: when `id_list` is not None, we can precompute the set of handles and then pass this set into Optimizer to use as `all_handles`.
I don't want to pursue this until we have a working implementation.

Reply:
Now that everything is working I can work on this, and the line 29 optimization if any set is empty. (I'll do some speed tests to see if that is worthwhile)

Reply:
Thanks for fixing the id_list speedup, and handling the tuple interface.